src/main/scala/scalismo/faces/image/pyramid/LaplacePyramid.scala

```diff
@@ -56,10 +60,20 @@ object LaplacePyramid {
   /**
     * Standard method to upscale an image.
     */
-  def expand[A: ClassTag](implicit ops: ColorSpaceOperations[A]) = new ImageFilter[A, A] {
-    override def filter(image: PixelImage[A]): PixelImage[A] = {
-      val w = image.width * 2
-      val h = image.height * 2
+  def expand[A: ClassTag](implicit ops: ColorSpaceOperations[A]) = new ExpandFilter[A] {
+    override def filter(image: PixelImage[A], upperWidth: Int, upperHeight: Int): PixelImage[A] = {
+      val w = {
+        val proposed = image.width * 2
```

Comment:
Yes good point! How much the image is scaled should only depend on upperWidth and upperHeight. The scaling of the image should only be determined by the reduce function supplied to the GaussPyramid.